src/lib.rs

```diff
@@ -169,15 +169,32 @@ pub struct Interpreter<'a> {
 }
 
 impl<'a> Interpreter<'a> {
-    fn new<R: Read, W: Write>(code: String, input: &'a mut R, output: &'a mut W) -> Interpreter<'a> {
-        Interpreter {
-            code: code,
+    /// Return a new interpreter with the given code, reader, and writter.
+    ///
+    /// Interpreters are relatively large, so avoid too many calls to this
+    /// function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use brainfuck::Interpreter;
+    ///
+    /// let mut reader = &[][..];
+    /// let mut writer = Vec::<u8>::new();
+    /// Interpreter::new("+.", &mut reader, &mut writer).unwrap().run();
+    /// ```
+    pub fn new<C: Into<String>, R: Read, W: Write>(code: C, input: &'a mut R, output: &'a mut W) -> Result<Interpreter<'a>, Error> {
+        // TODO: Compress and cache the code, removing everything but code.
+        //       This will allow running to avoid the overhead of finding
+        //       instructions and brace matching.
+        Ok(Interpreter {
```

Comment:
All I did was make `new` return a `Result` instead of an `Interpreter` directly. This is to be future compatible with the checking that I want `new` doing for things like unmatched brackets.